x/pairing/types/relay_exchange.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strings"
+
+	"github.com/lavanet/lava/utils"
+)
+
+type RelayExchange struct {
+	Request RelayRequest
+	Reply   RelayReply
+}
+
+func NewRelayExchange(req RelayRequest, res RelayReply) RelayExchange {
+	return RelayExchange{Request: req, Reply: res}
+}
+
+func (re RelayExchange) GetSignature() []byte {
+	return re.Reply.Sig
+}
+
+func (re RelayExchange) PrepareForSignature() []byte {
+	re.Reply.Sig = []byte{}
+	metadataBytes := make([]byte, 0)
+	for _, metadata := range re.Reply.GetMetadata() {
+		data, err := metadata.Marshal()
+		if err != nil {
+			utils.LavaFormatError("metadata can't be marshaled to bytes", err)
+		}
+		metadataBytes = append(metadataBytes, data...)
+	}
+	// we remove the salt from the signature because it can be different
+	re.Request.RelayData.Salt = []byte{}
+	return bytes.Join([][]byte{re.Reply.GetData(), []byte(re.Request.RelayData.String()), metadataBytes}, nil)
+}
+
+func (re RelayExchange) HashCount() int {
+	return 2
+}
+
+func (rp RelayPrivateData) GetContentHashData() []byte {
+	requestBlockBytes := make([]byte, 8)
+	metadataBytes := make([]byte, 0)
+	metadata := rp.Metadata
+	for _, metadataEntry := range metadata {
+		metadataBytes = append(metadataBytes, []byte(metadataEntry.Name+metadataEntry.Value)...)
+	}
+	binary.LittleEndian.PutUint64(requestBlockBytes, uint64(rp.RequestBlock))
+	addon := rp.Addon
+	if addon == nil {
+		addon = []string{}
+	}
+	msgData := bytes.Join([][]byte{metadataBytes, []byte(strings.Join(addon, "")), []byte(rp.ApiInterface), []byte(rp.ConnectionType), []byte(rp.ApiUrl), rp.Data, requestBlockBytes, rp.Salt}, nil)
```

Comment:
so here the salt is taken as is (unlike in RelayExchange, where it is reset)?

Reply:
yes. That's how the original implementation worked. In this PR I just "copied" the functionality of the old sign functions to new ones. I bet that it can be improved in both readability and performance if we sat with @omerlavanet to understand all the mechanisms in place